DatadogCore/Tests/Datadog/Tracing/TracingURLSessionHandlerTests.swift

```diff
@@ -227,4 +227,47 @@ class TracingURLSessionHandlerTests: XCTestCase {
             ]
         )
     }
+
+    func testGivenFirstPartyInterception_whenInterceptionStarts_withActiveSpan_itSendsParentSpanID() throws {
+        let core = DatadogCoreProxy()
+        var config = Trace.Configuration(
+            urlSessionTracking: Trace.Configuration.URLSessionTracking(
+                firstPartyHostsTracing: .traceWithHeaders(hostsWithHeaders: ["www.example.com": [.datadog]], sampleRate: 100)
+            )
+        )
+        config.traceIDGenerator = RelativeTracingUUIDGenerator(startingFrom: 1, advancingByCount: 1)
+        try Trace.enableOrThrow(with: config, in: core)
+
+        // Given
+        let request: URLRequest = .mockWith(url: "https://www.example.com")
+        let span = Tracer.shared(in: core).startRootSpan(operationName: "root")
+        let task: URLSessionTask = .mockWith(request: request, response: .mockResponseWith(statusCode: 200))
+
+        // When
+        span.setActive() // start root span
+        let feature = core.get(feature: NetworkInstrumentationFeature.self)
+        // simulate a URLSessionTask intereception
+        feature?.intercept(task: task, additionalFirstPartyHosts: .init(["https://www.example.com"]))
+        feature?.task(task, didFinishCollecting: .mockAny())
+        feature?.task(task, didCompleteWithError: nil)
+        feature?.flush()
+        //
+        span.finish() // finish root span
```

Comment:
blocker/ This test can work because both are called on the same thread:
- `span.setActive()`
- `feature.intercept(task:additionalFirstPartyHosts:)` 

How do we guarantee they will be called on the same thread in real use case? Note that `span` is user object and `intercept(task:additionalFirstPartyHosts:)` is an internal method in the SDK.

We must add integration test that uses high-level APIs (ideally only `public`), not a one that simulates behaviour by calling internal methods.

Reply:
We can't guarantee that the `span.setActive()` is called on the same thread. If it's not, then the parent won't be propagated and that's expected.

> Note that span is user object and intercept(task:additionalFirstPartyHosts:) is an internal method in the SDK.
Yes, I'm aware. For this test specifically, I'm following the pattern of this suite. But I can add a more high level test indeed.